scripts/node-sync.sh

```diff
@@ -39,36 +45,53 @@ injectived_backup() {
 
 injectived_sync() {
   echo $MSG_SYNC_START
-  aws s3 sync --no-sign-request --delete s3://injective-snapshots/mainnet/injectived/data ~/.injectived/data
+  mkdir temp && cd temp
+  URL=$(curl -L https://quicksync.io/injective.json | jq -r '.[] |select(.file=="injective-1-pruned")|.url')
+  aria2c -x5 $URL
+  lz4 -d $(basename $URL) | tar xf - -C $HOME/.injectived
+  cd .. && rm -rf temp
 }
 
 injectived_restart() {
   injectived version
   injectived start
-  #TODO: Update binary if required
+}
+
+injectived_wait() {
+  timeout $WAIT_TIMEOUT bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' http://localhost:26657/health)" != "200" ]]; do sleep 5; done' || {
+    echo "Chain readiness probe exceeded timeout"
+    exit 1
+  }
 }
 
 injectived_healthcheck() {
-  # check if it's syncing from height > 2000000
-  height = curl localhost:26657/status | grep height
-  if height >$CHAIN_SYNC_HEIGHT; then
-    $MSG_SYNC_SUCCESS
-  else
-    echo $MSG_SYN_FAILED
-  fi
+  COUNT=0
+  while true; do
+    [[ $COUNT -lt 5 ]] || break
+    LAST_HEIGHT=$(curl -sS http://localhost:26657/block | jq .result.block.header.height | tr -d '"')
+    sleep 5
+    CUR_HEIGHT=$(curl -sS http://localhost:26657/block | jq .result.block.header.height | tr -d '"')
+    if [[ $LAST_HEIGHT -lt $CUR_HEIGHT ]]; then
```

Comment:
Since we are pointing to the latest release and using a pruned snapshot that is updated daily. Checking if the node is syncing for a few blocks should suffice. We can also wait till it has fully synced but that seems excessive and time-consuming ( correct me if I'm wrong :)) )

Reply:
You can also periodically query our height oracle service (aka `horacle`), its source of truth regarding block height and network state. 

`curl https://k8s.mainnet.horacle.injective.network/height/latest`

Reply:
I'm having trouble visualizing how this would fit in. Could you kindly elaborate? Interested in knowing bec it might be a better solution :))

For context, this snippet validates that height x where the snapshot was taken could be synced from by checking if it catches up for a few blocks. (not till full sync)

I think comparing current block to the horacle would have the same effect as checking the `catching_up` flag. We can use this instead if we don't agree with the drawback I gave in my first comment.

Thanks!!